src/testUtils.jsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react'
+import { render } from '@testing-library/react'
+import { BrowserRouter } from 'react-router-dom'
+import { I18nextProvider } from 'react-i18next'
+import { SettingsProvider } from './context/SettingsContext'
+import { WalletProvider } from './context/WalletContext'
+import { WebsocketProvider } from './context/WebsocketContext'
+import i18n from './i18n/testConfig'
+
+const AllTheProviders = ({ children }) => {
+  return (
+    <BrowserRouter>
+      <I18nextProvider i18n={i18n}>
+        <SettingsProvider>
+          <WalletProvider>
+            <WebsocketProvider>{children}</WebsocketProvider>
+          </WalletProvider>
+        </SettingsProvider>
+      </I18nextProvider>
+    </BrowserRouter>
+  )
+}
+
+const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options })
+
+// re-export everything
+export * from '@testing-library/react'
+
+// override render method
+export { customRender as render }
```

Comment:
I understand that we do not want to repeat ourselves and keep things simple. But I cannot help but feel this might make troubles in the future. Some components may not need all providers - most certainly don't. 

I'd be in favor of naming it differently. However, not a hard no! What do you think?

Reply:
> > Some components may not need all providers - most certainly don't.
> 
> That is true. But nevertheless in the actual app, all components will be rendered [with all providers present](https://github.com/joinmarket-webui/joinmarket-webui/blob/master/src/index.jsx#L17) since everything is a child of `App.jsx`. So this might be closer to the real thing? I can still rename the function of course! but I would probably say let's keep using that renamed render function nevertheless.

I am with you. Nothing we couldn't easily change if we see the need to do so.
Still a bit unhappy about `WebsocketProvider` being in there for every component, as I feel this will slow down the tests. But lets us see if it even becomes a problem. I am good with keeping it that way.

Reply:
Yeah right, the websocket might be annoying. Let's see how it develops and act if necesary!